### PR TITLE
Sta 442 multiple details request

### PR DESCRIPTION
**Pull request by MisoMurin**
- new internal model classes for parsing API responses
- each internal model class has a `convert()` which output is an object of appropriate public model class
- api classes moved to `api` package
- method for requesting the details of multiple places - 1 response type for both places list request (witjout details) and detailed places request

The new model classes are "copy-pasted", only convert methods are added.

### CODE REVIEW

File: app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt [hunk 1]
```diff
@@ -72,14 +72,13 @@ class PlacesListActivity : AppCompatActivity() {
...
-        get() = object : PlacesAdapter.ViewHolder.PlaceClick {
-            override fun onPlaceClick(position: Int) {
-                val placeDetailIntent = Intent(this@PlacesListActivity, PlaceDetailActivity::class.java)
-                placeDetailIntent.putExtra(ID, places!![position].id)
-                startActivity(placeDetailIntent)
-            }
+    private val onPlaceClick = object: PlacesAdapter.ViewHolder.PlaceClick {
+        override fun onPlaceClick(position: Int) {
+            val placeDetailIntent = Intent(this@PlacesListActivity, PlaceDetailActivity::class.java)
+            placeDetailIntent.putExtra(ID, places!![position].id)
```
Comment (mholecy) [inline app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt]: Safer: `places?.get(position)?.id`

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/media/Attribution.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        attribution.author = author
+        attribution.authorUrl = authorUrl
+        attribution.license = license
+        attribution.licenseUrl = licenseUrl
+        attribution.other = other
+        attribution.title = title
+        attribution.titleUrl = titleUrl
+
+        return Attribution()
```
Comment (mholecy) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/media/Attribution.kt]: `return attribution`

File: app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt (hunk 1 above)
Comment (hrach) [inline app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt]: Well, then the activity gets null id and will crash anyway, doesn't it?

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt [hunk 3]
```diff
@@ -81,4 +83,53 @@ internal class PlaceDetail {
...
+		place.url = url
+		place.thumbnailUrl = thumbnailUrl
+		place.marker = marker
+		place.parentIds = parentIds
+		place.detail = convertDetail()
+		
+		return place
+	}
+
+	fun convertDetail(): Detail {
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt]: private?

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt [hunk 4]
```diff
@@ -81,4 +83,53 @@ internal class PlaceDetail {
...
+		val detail = Detail()
+		
+		val convertedTags: MutableList<com.sygic.travel.sdk.model.place.Tag> = mutableListOf()
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt]: import package

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt [hunk 5]
```diff
@@ -81,4 +83,53 @@ internal class PlaceDetail {
...
+		tags?.mapTo(convertedTags) {
+			it.convert()
+		}
+
+		val convertedReferences: MutableList<com.sygic.travel.sdk.model.place.Reference> = mutableListOf()
+		references?.mapTo(convertedReferences){
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt]: conding style, space before {

File: app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt (hunk 1 above)
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/list/PlacesListActivity.kt]: That's probable

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt (hunk 3 above)
Comment (MisoMurin) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt]: Yes, was thinking about it.

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt (hunk 4 above)
Comment (MisoMurin) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/PlaceDetail.kt]: We will rename all internal model classes in order to be distinguishable from their public model classes equivalents. So the package paths wouldn't be required, just imports.

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/geo/Bounds.kt [hunk 6]
```diff
@@ -17,4 +18,15 @@ internal class Bounds {
 
     @SerializedName("west")
     var west: Float = 0.toFloat()
+
```
Comment (mholecy) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/geo/Bounds.kt]: Is it good design practice to add some functionality to the model classes ?

Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/geo/Bounds.kt]: Generally, anemic model is considered as a bad practice, so it should be ok to put logic there, the question may be if this is the logic which should be here.

**Pull request closed without merging**